src/system/SystemPacketBuffer.h

```diff
@@ -592,9 +575,25 @@ class DLL_EXPORT PacketBufferHandle
      *
      * @note This should be used ONLY by low-level code interfacing with LwIP.
      */
-    struct pbuf * GetLwIPpbuf() { return static_cast<struct pbuf *>(mBuffer); }
+    struct pbuf * UnsafeGetLwIPpbuf() { return static_cast<struct pbuf *>(mBuffer); }
 #endif // CHIP_SYSTEM_CONFIG_USE_LWIP
 
+    /**
+     * Export a raw PacketBuffer pointer.
+     *
+     * @brief The PacketBufferHandle's ownership is transferred to the caller.
+     *
+     * @note This should only be used in low-level code, e.g. to export buffers from LwIP or a similar stack.
+     *       The ref-qualifier `&&` requires the caller to use `std::move` to emphasize that ownership is
+     *       moved out of this handle.
+     */
+    CHECK_RETURN_VALUE PacketBuffer * UnsafeRelease() &&
```

Comment:
This is currently only used to reference a PacketBuffer in events. In the case of `CHIPoBLEWriteReceived` it's as a `PacketBuffer*`, but in `ChipSystemLayerEvent` it's as a `uintptr_t`. An option would be to only Release as an opaque `uintptr_t` to discourage any use of packet buffers through a `PacketBuffer*`; it might then also be possible to make `PacketBuffer` itself a private type inside `PacketBufferHandle`.